src/main.rs

```diff
@@ -490,6 +490,13 @@ impl MainHandler {
                                 let mut data = field.data.readable().unwrap();
                                 let headers = &field.headers;
                                 let mut target_path = path.clone();
+                                
+                                if headers.filename.clone().unwrap().contains("../") {
+                                    return Err((
+                                        status::Forbidden,
+                                        format!("Using filename to elevate folder scope is forbidden."),
+                                    ));
+                                }
```

Comment:
Check if the final path is the sub-path of root_path is a better idea.

Reply:
I'm not familiar enough with the codebase or these libraries to find a clean way to do what you want. I will close this PR.